### PR TITLE
Add v1.52.1-1.52.3 release of grpc-go to interop matrix

### PR DESCRIPTION
**Pull request by arvindbr8**
Adding gRPC-go v1.52.3 to the interop matrix

### CODE REVIEW

**Comment by easwars** [comment]
@jtattermusch

**Comment by arvindbr8** [comment]
I started an adhoc interop matrix [run](https://fusion2.corp.google.com/invocations/c561cd99-cf1c-4f2e-826a-18eecf8fb668/targets)

File: tools/interop_matrix/client_matrix.py [hunk 1]
```diff
@@ -211,6 +211,7 @@ def __init__(self, patch=[], runtimes=[], testcases_file=None):
             ('v1.50.1', ReleaseInfo(runtimes=['go1.16'])),
             ('v1.51.0', ReleaseInfo(runtimes=['go1.16'])),
             ('v1.52.0', ReleaseInfo(runtimes=['go1.19'])),
+            ('v1.52.3', ReleaseInfo(runtimes=['go1.19'])),
```
Comment (jtattermusch) [inline tools/interop_matrix/client_matrix.py:214]: usually we'd only add a single patch version for a given minor release (e.g. we'd only have the latest patch from 1.52.x). I'd suggest removing 1.52.0 while adding 1.52.3.

**Review by jtattermusch — APPROVED** [review]
LGTM after addressing my comments and after the adhoc interop matrix run is green.
- only test the latest patch version
- please update the PR description to match reality

The previous run of "adhoc interop matrix" has failed, so I started it again (looks like previously the docker images weren't uploaded.

**Comment by arvindbr8** [comment]
Hi @jtattermusch. Thanks for the review. Could you merge this for me please?

**Pull request merged**